packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,585 @@
+                { "Ref": "AWS::Region" },
+                ".",
+                { "Ref": "AWS::URLSuffix" },
+                "/",
+                { "Ref": assetParam },
+                "/",
+                { "Fn::Select": [
+                  0,
+                  { "Fn::Split": [
+                    "||",
+                    { "Ref": assetParamKey },
+                  ]},
+                ]},
+                { "Fn::Select": [
+                  1,
+                  { "Fn::Split": [
+                    "||",
+                    { "Ref": assetParamKey },
+                  ]},
+                ]},
+              ]],
+            },
+            "Parameters": {
+              "MyBucketParameter": "some-magic-bucket-name",
+            },
+          },
+        },
+      },
+    });
+  });
+
+  test('templates with nested stacks that were not provided in the nestedStacks property are left unmodified', () => {
+    new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-two-children.json'),
+    });
+
+    expect(stack).toMatchTemplate(loadTestFileToJsObject('parent-two-children.json'));
+  });
+
+  test('getNestedStack() throws an exception when getting a resource that does not exist in the template', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-two-children.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(() => {
+      parentTemplate.getNestedStack('FakeStack');
+    }).toThrow(/Nested Stack with logical ID 'FakeStack' was not found/);
+  });
+
+  test('getNestedStack() throws an exception when getting a resource that exists in the template, but is not a Stack', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-two-children.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(() => {
+      const childTemplate = parentTemplate.getNestedStack('ChildStack').includedTemplate;
+      childTemplate.getNestedStack('BucketImport');
+    }).toThrow(/Resource with logical ID 'BucketImport' is not a CloudFormation Stack/);
+  });
+
+  test('getNestedStack() throws an exception when getting a resource that exists in the template, but was not specified in the props', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-two-children.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(() => {
+      parentTemplate.getNestedStack('AnotherChildStack').includedTemplate.getNestedStack('BucketImport');
```

Comment:
This is weird... this should fail for this, right? 

```ts
parentTemplate.getNestedStack('AnotherChildStack');
```

Why is the test specifying something else then....?

Reply:
good catch...I don't know why the rest of that expression was there